rcl/src/rcl/remap_impl.h

```diff
@@ -69,7 +69,7 @@ typedef struct rcl_remap_impl_t
  * \return `RCL_RET_BAD_ALLOC` if allocating memory failed, or
  * \return `RCL_RET_ERROR` if an unspecified error occurs.
  */
-RCL_LOCAL
+RCL_PUBLIC
 RCL_WARN_UNUSED
 rcl_ret_t
 rcl_remap_copy(
```

Comment:
why not moving this function to a public header?
I don't see why it should be in a private header.

Reply:
I think in that case the function will be exposed to the users. And based on the `remap.h` and `arguments.h` code, is not expected that the user will deal with the function

Reply:
The same applies to:
- https://github.com/ros2/rcl/blob/70eba5e788af95c0e9f0dfe71af9ae630a9d4d76/rcl/include/rcl/remap.h#L42
- https://github.com/ros2/rcl/blob/70eba5e788af95c0e9f0dfe71af9ae630a9d4d76/rcl/include/rcl/remap.h#L268

those functions have no external users and they are public.

I don't see why we should half expose an API.

Reply:
Then shouldn't those be LOCAL as well? IMO there shouldn't be public that's not expected to be used by external users, I might be wrong.

Reply:
> Then shouldn't those be LOCAL as well? IMO there shouldn't be public that's not expected to be used by external users, I might be wrong.

You shouldn't have anything LOCAL in a public header neither something public in a private header.
I don't see why this shouldn't be in a public header.
@hidmic for feedback.

Reply:
LOCAL in a public header is definitely wrong, your code will compile but fail to link. PUBLIC in a private header won't do harm, but it is unusual. I'd be inclined towards putting `rcl_remap_copy` in `remap.h`.